src/overlays/actors/ovl_Obj_Takaraya_Wall/z_obj_takaraya_wall.c

```diff
@@ -30,39 +30,438 @@ ActorInit Obj_Takaraya_Wall_InitVars = {
+    *row = ((s32)pos->x + (14 * 120)) / 120;
+    *column = (s32)pos->z / 120;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Takaraya_Wall/func_80AD92FC.s")
+    if ((*row < 0) || (*row >= ARRAY_COUNT(sWallHeights)) || (*column < 0) ||
+        (*column >= ARRAY_COUNT(sWallHeights[0]))) {
+        return false;
+    }
+    return true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Takaraya_Wall/func_80AD9358.s")
+/*
+ * Checks if the given row and coulmn is valid and is open, i.e does not raise up.
+ */
+s32 ObjTakarayaWall_SpaceOpen(s32 row, s32 column) {
+    if ((row < 0) || (row >= ARRAY_COUNT(sWallHeights)) || (column < 0) || (column >= ARRAY_COUNT(sWallHeights[0])) ||
+        (sWallHeights[row][column] == 0.0f)) {
+        return false;
+    }
+
+    return true;
+}
+
+/*
+ * Checks nearby spaces if the adjacent space (which depends on the mode) or any of the three adjacent spaces to that
+ * space are closed (i.e the wall rises up).
+ * The modes check for the adjacent space towards the
+ *  0: Back (towards the spawned chest)
+ *  1: Right
+ *  2: Front
+ *  3: Left
+ * of the supplied space.
+ */
+s32 ObjTakarayaWall_NearbySpacesClosed(s32 row, s32 column, s32 mode) {
+    s32 row1;
+    s32 column1;
+    s32 adjacentSpaceRow;
+    s32 adjacentSpaceColumn;
+    s32 row2;
+    s32 column2;
+    s32 row3;
+    s32 column3;
+
+    if ((mode == 2) || (mode == 0)) {
+        if (mode == 0) {
+            adjacentSpaceRow = row - 1;
+            row3 = row - 2;
+        } else {
+            adjacentSpaceRow = row + 1;
+            row3 = row + 2;
+        }
+        adjacentSpaceColumn = column;
+        column3 = column;
+        row1 = adjacentSpaceRow;
+        column1 = column + 1;
+        row2 = adjacentSpaceRow;
+        column2 = column - 1;
+    } else {
+        if (mode == 1) {
+            adjacentSpaceColumn = column - 1;
+            column3 = column - 2;
+        } else {
+            adjacentSpaceColumn = column + 1;
+            column3 = column + 2;
+        }
+        adjacentSpaceRow = row;
+        row3 = row;
+        row1 = row + 1;
+        column1 = adjacentSpaceColumn;
+        row2 = row - 1;
+        column2 = adjacentSpaceColumn;
+    }
+
+    if ((adjacentSpaceRow < 0) || (adjacentSpaceRow >= ARRAY_COUNT(sWallHeights)) || (adjacentSpaceColumn < 0) ||
+        (adjacentSpaceColumn >= ARRAY_COUNT(sWallHeights[0])) ||
+        (sWallHeights[adjacentSpaceRow][adjacentSpaceColumn] == -10.0f) || ObjTakarayaWall_SpaceOpen(row1, column1) ||
+        ObjTakarayaWall_SpaceOpen(row2, column2) || ObjTakarayaWall_SpaceOpen(row3, column3)) {
+        return false;
+    }
+
+    return true;
+}
+
+void ObjTakarayaWall_BuildPath(s32 row, s32 column) {
+    s32 closedFlags = 0;
+    s32 closedCount = 0;
+    s32 mode;
+    s32 shift;
+
+    if (ObjTakarayaWall_NearbySpacesClosed(row, column, 0)) {
+        closedCount++;
+        closedFlags |= 1;
+    }
```

Comment:
I would usually suggest defines for the flags of `closedFlags`, but the way the code is using them is weird, so idk if it is worth

Reply:
Actually they make sense with the mode enum (though I've renamed it to direction as it actually makes sense in a few places).